cvprac/cvp_client.py

```diff
@@ -234,9 +227,19 @@ def _create_session(self, all_nodes=False):
         self.error_msg = '\n'
         for _ in range(0, num_nodes):
             host = next(self.node_pool)
-            self.url_prefix = ('%s://%s:%d/web' %
-                               (self.protocol, host, self.port))
+            self.url_prefix = ('https://%s:%d/web' % (host, self.port or 443))
             error = self._reset_session()
+            if error and not self.cert:
+                self.log.warning('Failed to connect over https. Potentially'
+                                 ' due to an old version of CVP. Attempting'
+                                 ' fallback to http. Error: %s' % error)
+                # Attempt http fallback if no cert file is provided. The
+                # intention here is that a user providing a cert file
+                # forces https.
+                self.url_prefix = self.url_prefix.replace('https', 'http', 1)
```

Comment:
I am still allowing the user to provide a port number and this would override that portion of the prefix.